src/ccb/ccb_server.cpp

```diff
@@ -984,15 +972,13 @@ CCBServer::AddTarget( CCBTarget *target )
 			continue;
 		}
 
-		if( m_targets.insert(target->getCCBID(),target) == 0 ) {
+		if (!m_targets.contains(target->getCCBID())) {
+			m_targets.emplace(target->getCCBID(), target);
 			EpollAdd(target);
 			break; // success
 		}
 
-		CCBTarget *existing = NULL;
-		if( m_targets.lookup(target->getCCBID(),existing) != 0 ) {
-				// That's odd: there is no conflicting ccbid, so why did
-				// the insert fail?!
+		if (m_targets.contains(target->getCCBID())) {
```

Comment:
This check looks backwards. The code wants to explode if no entry is found. I don't think this check is required at all. We should be able to trust our data structure's previous answer that there's an entry for this key.